webcam-capture/src/main/java/com/github/sarxos/webcam/WebcamMotionDetector.java

```diff
@@ -401,12 +401,14 @@ public void setDne(Rectangle rectangle) {
    */
   public ArrayList<Integer> getThresholds()
   {
-    if (detectorAlgorithm instanceof WebcamMotionDetectorDefaultAlgorithm
-        || detectorAlgorithm instanceof WebcamMotionDetectorDefaultWithDNE)
+    if (detectorAlgorithm instanceof WebcamMotionDetectorDefaultAlgorithm)

```

Comment:
Hi @kkanzelmeyer,

This code is of course valid, but it would be better to have getThreasholds() extracted to an interface, such as:

```java
public interface ThresholdsAlgorithm {
  ArrayList<Integer> getThresholds();
}
```

And then, both WebcamMotionDetectorDefaultAlgorithm and WebcamMotionDetectorDefaultWithDNE can implement this interface. With this we can change this code to check instanceof interface instead of every class:

```java
public ArrayList<Integer> getThresholds()
  if (detectorAlgorithm instanceof ThresholdsAlgorithm) {
    return ((ThresholdsAlgorithm) detectorAlgorithm).getThresholds();
  }
  throw new UnsupportedOperationException("This method is only valid for algorithms which implements " + ThresholdsAlgorithm.class);
}
```